File: 백준/Silver/1969. DNA/DNA.py
import sys
from collections import deque

N, M = map(int,sys.stdin.readline().strip().split())

dna = deque()
result = ""
dna_a = deque([0, 0, 0, 0])
distance = 0
max = 0
index = 0

for i in range(N):
  dna.append(list(sys.stdin.readline().strip()))



for i in range(M):
  for j in range(N):
    if dna[j][i] == "A": dna_a[0]+=1
    elif dna[j][i] == "C": dna_a[1]+=1
    elif dna[j][i] == "G": dna_a[2]+=1
    elif dna[j][i] == "T": dna_a[3]+=1
  for i in range(4):
    if max < dna_a[i]:
      max = dna_a[i]
      index = i
  distance += N-max
  if index==0: result += "A"
  elif index==1: result += "C"
  elif index==2: result += "G"
  elif index==3: result += "T"
  dna_a = deque([0, 0, 0, 0])
  max = 0
  
print(result)
print(distance)

File: 백준/Silver/1969. DNA/DNA.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.lang.reflect.Array;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
        String[] s = bf.readLine().split(" ");
        StringBuilder sb = new StringBuilder();

        int N = Integer.parseInt(s[0]);
        int M = Integer.parseInt(s[1]);
        int resultCount = 0;

        String[] arr = new String[N];
        for(int i=0; i<N; i++){
            arr[i] = bf.readLine();
        }

        for(int i=0; i<M; i++){
            int[] result = new int[4];
            for(int j=0; j<N; j++){
                char c = arr[j].charAt(i);
                if(c == 'A'){
                    result[0]++;
                }else if(c == 'C'){
                    result[1]++;
                }else if(c == 'G'){
                    result[2]++;
                }else{
                    result[3]++;
                }
            }
            int index = 0;
            int max = 0;
            for(int j=0; j<4; j++){
                if (max < result[j]){
                    index = j;
                    max = result[j];
                }
            }

            if(index == 0){
                sb.append('A');
            }else if(index == 1){
                sb.append('C');
            }else if(index == 2){
                sb.append('G');
            }else {
                sb.append('T');
            }
        }

        System.out.println(sb.toString());
        for(int i=0; i<N; i++){
            for(int j=0; j<M; j++){
                if(sb.charAt(j) != arr[i].charAt(j)){
                    resultCount++;
                }
            }
        }
        System.out.println(resultCount);
    }
}